hazelcast/src/main/java/com/hazelcast/jet/impl/execution/ProcessorTasklet.java

```diff
@@ -574,8 +574,8 @@ private void fillInbox() {
                 continue;
             }
             result = currInstream.drainTo(addToInboxFunction);
-            assert inbox.queue().stream().allMatch(i -> i instanceof SpecialBroadcastItem)
-                    || inbox.queue().stream().noneMatch(i -> i instanceof SpecialBroadcastItem)
+            assert inbox.queue().stream().allMatch(SpecialBroadcastItem.class::isInstance)
```

Comment:
> Method or constructor references are more readable than lambda expressions *in many situations*

I think this might be the case of "other situations"

Reply:
633e8880ae0f1fd2f696ef8d97b841ae000778f8

Happy to revert this if we can address the rest :)

Reply:
there are other similar `instanceof`s

Reply:
add78dcd56217a801f44aaed80f6df8926a48172

Reply:
@Patras3 I don't think there's a way to exclude `instanceof` from this rule in Sonar, I prefer the method reference but I can see how `instanceof` is more typical just because of the name.

Is it worth turning the rule off to avoid false positives going forwards?

Reply:
Right, there is no option to exclude `instanceof `from this rule 😞 For other code it mostly makes sense. 
Because rule has Minor severity and type 'Code Smell, I think we can ignore such cases (with instanceOf) from Sonar side to reduce noise but preserve warnings as it can help with other cases. WDYT ?

Reply:
> Right, there is no option to exclude `instanceof `from this rule 😞 For other code it mostly makes sense. Because rule has Minor severity and type 'Code Smell, I think we can ignore such cases (with instanceOf) from Sonar side to reduce noise but preserve warnings as it can help with other cases. WDYT ?

I'm easy either way. I think disabling the rule probably means less admin going forwards though rather than manually ignoring them?

Reply:
Ok, Let's wait for next analysis (it will be done during tonight), to check what's left and we decide.